### PR TITLE
Add support for private CAPZ E2E tests

### PR DESCRIPTION
**Pull request by tuladhar**
Towards https://github.com/giantswarm/giantswarm/issues/30767

### What this PR does
- Adds support for private CAPZ WC tests

### Checklist

- [x] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/run cluster-test-suites

### CODE REVIEW

**Review by AverageMarcus — COMMENTED** [review]
Is there a reason you didn't follow the same approach as CAPA and add a private cluster builder into cluster-standup-teardown? (e.g. https://github.com/giantswarm/cluster-standup-teardown/blob/main/pkg/clusterbuilder/providers/capa/private_cluster.go)

File: internal/common/apps.go [hunk 1]
```diff
@@ -31,6 +31,8 @@ func RunApps() {
 			defaultAppsAppName := fmt.Sprintf("%s-%s", state.GetCluster().Name, "default-apps")
 
 			if skipDefaultAppsApp {
+				// FIX ME: For private CAPZ tests, comment out the log line below.
+				// It cause a runtime error: invalid memory address or nil pointer dereference. Yet to know why.
```
Comment (AverageMarcus) [inline internal/common/apps.go:35]: ???

**Review by AverageMarcus — CHANGES_REQUESTED** [review]

**Comment by tuladhar** [comment]
> Is there a reason you didn't follow the same approach as CAPA and add a private cluster builder into cluster-standup-teardown? (e.g. https://github.com/giantswarm/cluster-standup-teardown/blob/main/pkg/clusterbuilder/providers/capa/private_cluster.go)

Sorry, I wasn't aware of the approach used in CAPA and these changes came directly from [Erkan’s previous PR](https://github.com/giantswarm/cluster-test-suites/pull/282). I agree, we should align with what other providers are doing. I’ll follow the same approach and make the necessary changes. Thanks for the suggestion!

File: internal/common/apps.go (hunk 1 above)
Comment (tuladhar) [reply internal/common/apps.go:35]: I yet don't know why it's trigging the panic, but will figure out and let you know.

**Comment by tuladhar** [comment]
Failed tests was for CAPV/CAPVCD provider, unrelated.

**Comment by tuladhar** [comment]
/run cluster-test-suites TARGET_SUITES=./providers/capz/private

**Review by AverageMarcus — APPROVED** [review]

**Pull request merged**